homeassistant/components/integration/sensor.py

```diff
@@ -174,22 +174,25 @@ def _unit(self, source_unit: str) -> str:
     async def async_added_to_hass(self) -> None:
         """Handle entity which will be added."""
         await super().async_added_to_hass()
-        if state := await self.async_get_last_state():
+        if (state := await self.async_get_last_state()) is not None:
             try:
-                self._state = Decimal(state.state)
+                if state.state not in (STATE_UNKNOWN, STATE_UNAVAILABLE):
+                    self._state = Decimal(state.state)
+                else:
+                    self._attr_available = False
+                    self._state = None
```

Comment:
My intention is to keep the error message in the except, so I think it is better to keep it separate ?

Reply:
Why would move it to `except` change anything with the error message?

Reply:
Btw; I think you should use `RestoreSensor` and not `RestoreEntity`

Reply:
Well, only the `Decimal()` call can throw, so that is the only thing that should be in this try/except block. All the rest of the logic needs to be taken out.